CIP-XXX/CIP-XXX.md

```diff
@@ -27,10 +27,53 @@ As part of this, we also describe the bit ordering within a `BuiltinByteString`,
 and provide some laws these operations should obey.
 
 ## Motivation: why is this CIP necessary?
+Currently, Plutus Core has a `BuiltinByteString` type, which is a packed array of bytes, similar to Haskell's ByteString.
+We want to implement bit wise operations on `BuiltinByteString` type. In otherwords, we want to achieve bit level granularity on the operations on `BuiltinByteString`.
+Some of the operations that we are interested in are, bit level OR, AND, NOT, or XOR. There are many added advantage of having bit level granularity which we will discuss in more detail later. Unfortunately, all oprations available on the `BuiltinByteString` works on the byte level granularity. So we need this CIP to bridge the gap.
+To explain the usefulness of this granulaity to manipulate `BuiltinByteString`, we use the following example of integer sets.
```

Comment:
There will be more examples: this sentence should lead into _all_ of them, not just one of them.